hdf5-sys/Cargo.toml

```diff
@@ -23,12 +23,12 @@ hdf5-src = { workspace = true, optional = true }
 # Please see README for further explanation of these feature flags
 [features]
 default = []
-mpio = ["mpi-sys"]
-hl = ["hdf5-src/hl"]
-threadsafe = ["hdf5-src/threadsafe"]
-zlib = ["libz-sys", "hdf5-src/zlib"]
-static = ["hdf5-src"]
-deprecated = ["hdf5-src/deprecated"]
+mpio = ["dep:mpi-sys"]
+hl = ["hdf5-src?/hl"]
+threadsafe = ["hdf5-src?/threadsafe"]
+zlib = ["libz-sys", "hdf5-src?/zlib"]
+static = ["dep:hdf5-src"]
+deprecated = ["hdf5-src?/deprecated"]
```

Comment:
It's the other way around, _**no**_ deprecated symbols.

So it should probably be called `no-deprecated`?

Reply:
So let's rename `deprecated` -> `no-deprecated` while we're changing things here?

Reply:
In `hdf5-src` we ask for the inclusion of deprecated symbols, I would like to keep this up the chain since asking for no deprecated symbols seems rather counter-intuitive.

Reply:
Ok, I got a bit confused myself :) But then... shouldn't we be checking `!header.have_no_deprecated` as opposed to `header.have_no_deprecated`?

Reply:
Yeah, you are right! Bit confusing with the double negation but should be correct now